starter-code/pizza.js

```diff
@@ -33,53 +48,176 @@ function renderEverything() {
 }
 
 function renderPepperonni() {
-  document.querySelectorAll('.pep').forEach(function($pep){
+  document.querySelectorAll('.pep').forEach(function ($pep) {
     if (state.pepperonni) {
       $pep.style.visibility = "visible";
-    }
-    else {
+    } else {
       $pep.style.visibility = "hidden";
     }
   })
 }
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll(".mushroom").forEach(function ($mush) {
+    if (state.mushrooms) {
+      $mush.style.visibility = "visible";
+    } else {
+      $mush.style.visibility = "hidden";
+    }
+  })
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll(".green-pepper").forEach(function ($pepper) {
+    if (state.greenPeppers) {
+      $pepper.style.visibility = "visible";
+
+    } else {
+      $pepper.style.visibility = "hidden"
+    }
+  })
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll(".sauce").forEach(function ($sauce) {
+    if (state.whiteSauce) {
+      $sauce.classList.add("sauce-white")
+    } else {
+      $sauce.classList.remove("sauce-white")
+    }
+  })
+
+
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  document.querySelectorAll(".crust").forEach(function ($crust) {
+    if (state.glutenFreeCrust) {
+      $crust.classList.add("crust-gluten-free")
+    } else {
+      $crust.classList.remove("crust-gluten-free")
+    }
+  })
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  if (state.pepperonni) {
```

Comment:
The main area where you could have cleaned it up a little btter was here:
you had the right  idea in the renderPrice function by using:
`state[element]`
In this case you had to hard code state.pepperonni and all the other cases.

Still works, though :)